src/main/kotlin/pt/up/fe/ni/website/backend/service/AccountService.kt

```diff
@@ -18,6 +21,10 @@ class AccountService(private val repository: AccountRepository, private val enco
 
         val account = dto.create()
         account.password = encoder.encode(dto.password)
+
+        val fileName = "${UUID.randomUUID()}.${dto.photoFile?.filenameExtension() ?: ""}"
```

Comment:
We could probably define photoFile as lateInit to get rid of those null checks. Might cause an exception if someone forgets it (and I'm not sure if it's actually a good thing)

Reply:
Do you remember the hell we had when fixing members' images? What if we preprend their name before the random id?

Reply:
> We could probably define photoFile as lateInit to get rid of those null checks. Might cause an exception if someone forgets it (and I'm not sure if it's actually a good thing)

I think photofile can be null since the photo is not a required field, so I think this is unnecessary

> Do you remember the hell we had when fixing members' images? What if we preprend their name before the random id?

Yes this seems like a good idea